### PR TITLE
expandable and generic bridge adapter

### PR DESCRIPTION
**Pull request by joshrrose**
## Description

Expandable bridge adapter and made it generic. Removed the chain specific BridgeAdapters. 

## Checklist:

-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Review by kyu-c — COMMENTED** [review]
The approach makes sense to me! ❤️ 

Have we considered using single BridgeAdapter (and the same groups) for all chains? (that was the old model) 🤔 

Pros:
* When we add a new protocol, we only need to modify one adapter (vs N adapters).
* Simpler deployment scripts 
* When an existing protocol expands to a different chain, no deployment is needed. 

Cons:
* Costs more gas on deployments.

**Comment by joshrrose** [comment]
@kyu-c I had not thought of that. That might be an option but would require changes to 0x api because the protocol Ids would change wouldn't the?

**Comment by kyu-c** [comment]
> @kyu-c I had not thought of that. That might be an option but would require changes to 0x api because the protocol Ids would change wouldn't the?

I believe the same protocol ids are used across all. A protocol id is just an enum value of [BridgeProtocols.sol](https://github.com/0xProject/protocol/blob/3f8dbf3fe900a8b07a0f915f4b89bd2af6b3ecbe/contracts/zero-ex/contracts/src/transformers/bridges/BridgeProtocols.sol#L29-L61)

**Comment by joshrrose** [comment]
Indeed. I think the blocker might be "weth" then. The bridge adapters get a single argument of the wrapped native token which would require some refactoring. I'll start working on the other chains with the above method. If at some point we want to merge them all we can but will need to figure out how to deal with "weth" for multiple chains

File: contracts/zero-ex/test/artifacts.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import * as ERC1155OrdersFeature from '../test/generated-artifacts/ERC1155Orders
 import * as ERC165Feature from '../test/generated-artifacts/ERC165Feature.json';
 import * as ERC721OrdersFeature from '../test/generated-artifacts/ERC721OrdersFeature.json';
 import * as EthereumBridgeAdapter from '../test/generated-artifacts/EthereumBridgeAdapter.json';
+import * as EthereumBridgeAdapterGroup1 from '../test/generated-artifacts/EthereumBridgeAdapterGroup1.json';
```
Comment (kyu-c) [inline contracts/zero-ex/test/artifacts.ts:21]: nit: obsolete code here and below.

**Review by kyu-c — COMMENTED** [review]
Draft LGTM

File: contracts/zero-ex/contracts/src/transformers/bridges/BridgeAdapterGroup2.sol [hunk 2]
```diff
@@ -50,42 +54,37 @@ contract FantomBridgeAdapter is
...
             boughtAmount = _tradeWOOFi(sellToken, buyToken, sellAmount, order.bridgeData);
-        } else if (protocolId == BridgeProtocols.UNKNOWN) {
+        } else if (protocolId == BridgeProtocols.AAVEV3) {
             if (dryRun) {
                 return (0, true);
             }
-            boughtAmount = _tradeZeroExBridge(sellToken, buyToken, sellAmount, order.bridgeData);
-        }
+            boughtAmount = _tradeAaveV3(sellToken, buyToken, sellAmount, order.bridgeData);
+        }
```
[the remark this answers is not in the document]
Comment (joshrrose) [reply contracts/zero-ex/contracts/src/transformers/bridges/BridgeAdapterGroup2.sol:87]: I think we would have to discuss with Jacob and the rest of Protocol. Reverts don't currently get bubbled up. Something I found while attempting to debug. I had to add some code to get the actual reverts to show up. I believe part of the reason is to allow for fallback sources, though i don't know if we actually use this feature anymore

**Pull request closed without merging**